src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -531,11 +531,12 @@ void WindowImplX11::setIcon(unsigned int width, unsigned int height, const Uint8
 
     xcb_generic_error_t* errptr = xcb_request_check(m_connection, cookie);
     bool setWindowIconFailed = (errptr != NULL);
+    uint8_t error_code = errptr->error_code;
```

Comment:
I was confused above and my comment was useless so I removed it.

According to the docs it either returns an error or `NULL` as such we should always be able to retrieve an `error_code`.